hw3/ThronesAPI/src/HousesChart.jsx

```diff
@@ -0,0 +1,116 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import { Chart as ChartJS, ArcElement, Tooltip, Legend } from "chart.js";
+import { Doughnut } from "react-chartjs-2";
+import { Container } from "react-bootstrap";
+import NavbarComponent from "./NavbarComponent";
+
+ChartJS.register(ArcElement, Tooltip, Legend);
+
+const backgroundColors = [
+  "rgba(54, 162, 235, 0.8)",
+  "rgba(255, 206, 86, 0.8)",
+  "rgba(255, 99, 132, 0.8)",
+  "rgba(75, 192, 192, 0.8)",
+  "rgba(153, 102, 255, 0.8)",
+  "rgba(255, 159, 64, 0.8)",
+  "rgba(199, 199, 199, 0.8)",
+  "rgba(83, 102, 255, 0.8)",
+  "rgba(40, 159, 64, 0.8)",
+  "rgba(210, 199, 199, 0.8)",
+  "rgba(78, 52, 199, 0.8)",
+];
+
+const borderColors = [
+  "rgba(54, 162, 235, 1)",
+  "rgba(255, 206, 86, 1)",
+  "rgba(255, 99, 132, 1)",
+  "rgba(75, 192, 192, 1)",
+  "rgba(153, 102, 255, 1)",
+  "rgba(255, 159, 64, 1)",
+  "rgba(159, 159, 159, 1)",
+  "rgba(83, 102, 255, 1)",
+  "rgba(40, 159, 64, 1)",
+  "rgba(210, 199, 199, 1)",
+  "rgba(78, 52, 199, 1)",
+];
+
+const url = "https://thronesapi.com/api/v2/Characters";
+
+const HousesChart = () => {
+  const [houseCounts, setHouseCounts] = useState([]);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        const response = await axios.get(url);
+        const apiData = response.data;
+
+        const allFamilies = apiData.map((character) => character.family);
+        allFamilies.forEach((house, index) => {
+          if (house.includes("Lanister")) {
+            allFamilies[index] = "House Lannister";
+          }
+        });
```

Comment:
Nice way of doing this. Will this take care of all the typos in house names? If so I would want to implement
something similar in my data cleanup for this exercise.

Reply:
Thanks! I have added if-else conditions and only Lannister was not getting considered in the typos. Including this now looks like everything is considered.